### PR TITLE
feat: ruby template

### PR DESCRIPTION
**Pull request by samrose**
Ruby template

### CODE REVIEW

File: templates/ruby/pkgs/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  };
+in stdenv.mkDerivation rec {
+  name = "my-package";
+  src = ../.;
+  buildInputs = [gems ruby];
+  installPhase = ''
+    mkdir -p $out/bin  $out/share/${name}
+    cp -r * $out/share/${name}
+    bin=$out/bin/${name}
+# we are using bundle exec to start in the bundled environment
```
Comment (mkenigs) [inline templates/ruby/pkgs/default.nix:19]: Not sure if this indentation is intentional?

File: templates/ruby/pkgs/default.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
+{ stdenv, bundlerEnv, ruby }:
+let
+  # the magic which will include gemset.nix
+  gems = bundlerEnv {
+    name = "my-package-env";
+    inherit ruby;
+    gemfile = ../Gemfile;
+    lockfile = ../Gemfile.lock;
+    gemset = ../gemset.nix;
```
Comment (mkenigs) [inline templates/ruby/pkgs/default.nix:9]: I think you said you used bundix? Will that be commented or documented somewhere?

File: templates/ruby/pkgs/default.nix (hunk 1 above)
Comment (samrose) [reply templates/ruby/pkgs/default.nix:19]: @mkenigs thanks I will clean this up as soon as I am in front of dev computer again

File: templates/ruby/pkgs/default.nix (hunk 2 above)
Comment (samrose) [reply templates/ruby/pkgs/default.nix:9]: @mkenigs yes for sure. I'll make special mention of it + the steps to follow in the docs. I'll actually add a corresponding comment here too. Thanks!

Comment (samrose) [reply templates/ruby/pkgs/default.nix:12]: pushed a change now to resolve

Comment (mkenigs) [reply templates/ruby/pkgs/default.nix:12]: Hmm just seeing the formatting change, did you add something for this?

Comment (samrose) [reply templates/ruby/pkgs/default.nix:12]: Still need to add the comment on but fix. Was referring to formatting

Comment (samrose) [reply templates/ruby/pkgs/default.nix:13]: Comment added

**Review by mkenigs — APPROVED** [review]

**Comment by tomberek** [comment]
```
gemfile = ../Gemfile;
lockfile = ../Gemfile.lock;
gemset = ../gemset.nix;
```
It's fine to expect Ruby users to understand the pointers to Gemfile and Gemfile.lock, but will need guidance on gemset.nix.

**Comment by mkenigs** [comment]
> It's fine to expect Ruby users to understand the pointers to Gemfile and Gemfile.lock, but will need guidance on gemset.nix.

He has
```
  # running the bundix command
  # will generate the gemset.nix file below
```

Does that need to be more verbose? Or moved to right above the gemset line?

**Comment by samrose** [comment]
per @limeytexan I am closing this PR as it is already covered under the `mkDerivation` (currently named "generic") template

**Comment by mkenigs** [comment]
Should we just call it the `mkDerivation - Ruby` template? And if not, can we at least keep your work in a flox-examples Ruby example?

**Comment by samrose** [comment]
> Should we just call it the `mkDerivation - Ruby` template? And if not, can we at least keep your work in a flox-examples Ruby example?

@mkenigs a good question. I also ended up changing the docs so that they make sense with naming things after the builder instead of the language. In this case, and in the Java/Apache Ant case, I guess we could make notes under the new "mkDerivation" tab about nuances with languages like C, Ruby, and Java where Ant builder is used.

**Comment by mkenigs** [comment]
I just think if I'm a Nix newcomer, having it at the top of the tab would be helpful. I would have no idea that I should click on mkDerivation if I want to write something in Ruby

**Comment by samrose** [comment]
> I just think if I'm a Nix newcomer, having it at the top of the tab would be helpful. I would have no idea that I should click on mkDerivation if I want to write something in Ruby

I tend to agree. @limeytexan requested that things appear this way, so this is my fulfillment of that request.

**Comment by limeytexan** [comment]
Thanks @mkenigs for highlighting, I agree this is somewhat of an exception, because in effect the "builder" for ruby gems appears to be the _combination_ of `bundlerEnv` and `mkDerivation`, which isn't handled by the other pure `mkDerivation` case.

Perhaps we should call this template "bundlerEnv+mkDerivation", or perhaps "mkDerivation-ruby"?

I think I like the idea of adopting the following naming scheme:
* by default, use the builder name, e.g. "mkDerivation"
* if a builder is used for different languages or purposes then we can append a relevant suffix, e.g. "mkDerivation-ruby" or "mkDerivation-java"

The above is a weakly-held opinion - happy to entertain all reasonable suggestions!

**Comment by limeytexan** [comment]
Oops ... meant to reopen.

**Pull request merged**